src/components/SettlementButton/index.tsx

```diff
@@ -67,6 +67,8 @@ function SettlementButton({
     const [policy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, {canBeMissing: false});
     const isInvoiceReport = (!isEmptyObject(iouReport) && isInvoiceReportUtil(iouReport)) || false;
     const {isAccountLocked, showLockedAccountModal} = useContext(LockedAccountContext);
+    const [allSnapshots] = useOnyx(ONYXKEYS.COLLECTION.SNAPSHOT, {canBeMissing: true});
+    const transactions = getReportTransactions(iouReport?.reportID);
```

Comment:
We should remove `getReportTransactions` and use `useOnyx` instead. Otherwise, we'll have the same issue where if the report transaction is updated, the component won't re-render. We should use a selector to get only the transactions we care about, e.g. https://github.com/Expensify/App/blob/e2c82c0e0242568d7a17fda5dfb84ed8635506ee/src/components/MoneyReportHeader.tsx#L171-L175